### PR TITLE
(Doc) Add Deprecated note to Associate Profile section in doc

### PR DESCRIPTION
**Pull request by IgorCATech**
Signed-off-by: ik673626 <igor.kazmyr@broadcom.com>

## Proposed changes

This PR addresses the #1572 issue. It adds a note to the Associate Profile feature description. The note notifies a user that the feature is deprecated and with the release of V2 will be entirely removed.

## Release Notes

Milestone: should have been 1.18

Changelog: 

## Types of changes

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Updates to Documentation or Tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/vscode-extension-for-zowe/wiki/Best-Practices:-Contributor-Guidance) wiki
- [x] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [x] PR Description is included
- [ ] gif or screenshot is included if visual changes are made
- [ ] `yarn workspace vscode-extension-for-zowe vscode:prepublish` has been executed
- [ ] All checks have passed (DCO, Jenkins and Code Coverage)
- [ ] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [ ] There is coverage for the code that I have added
- [ ] I have tested it manually and there are no regressions found
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any PR dependencies have been merged and published (if appropriate)

## Reminder

After a PR is merged into the `master` branch, create a PR from `master` to the `next` branch resolving any conflicts.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by IgorCATech** [comment]
Just to reiterate something I've mentioned in the stand-up today. Please advise whether it's a good idea to add information about the V2 release date plan. I added it with my latest commit, but maybe this is something we don't want to disclose. It's more of a product/marketing question, I suppose.

**Comment by IgorCATech** [comment]
Any comments/objections so far?

**Review by crawr — COMMENTED** [review]
I think `removed in version 2` reads better to me.

File: packages/zowe-explorer/README.md
Comment (jellypuno) [inline packages/zowe-explorer/README.md:362]: @IgorCATech Just re-direct them [here](https://www.zowe.org/download.html) for the release timeline

**Review by lauren-li — DISMISSED** [review]
This looks good to me! Thanks @IgorCATech for the update!

**Review by crawr — DISMISSED** [review]
LGTM! Thanks @IgorCATech

File: packages/zowe-explorer/README.md
Comment (IgorCATech) [reply packages/zowe-explorer/README.md:362]: As @lauren-li suggested, and I completely support the suggestion, it would be good to add the anchor link that lands on the appropriate heading on that page. I'll need to add an ID to that heading to be able to accomplish this task. We can either wait for me to do that, then merge this PR, or merge this PR first and, once the HTML page on the Zowe website is fixed, create a new PR with the small fix. 
If this Deprecated note can wait, I will go for the former option. Let me know what works better for you. Thanks

Comment (lauren-li) [reply packages/zowe-explorer/README.md:362]: @IgorCATech Unless the Zowe website will be updated soon, I would go ahead and merge in this PR and then create another PR when the fix is in. It is more important that we give any users of Associated Profiles a timely deprecation warning, and plus we already describe in this PR update what section of the Download page to look for.

The anchor link to the release timeline section will be a nice enhancement to increase convenience for our users when it is ready. Thanks for looking into it!

Comment (IgorCATech) [reply packages/zowe-explorer/README.md:362]: https://github.com/zowe/zowe.github.io/pull/271 - PR opened

File: packages/zowe-explorer/README.md
Comment (lauren-li) [inline packages/zowe-explorer/README.md:359]: @IgorCATech This is a bit nitpicky and not technically part of the PR, but could we change "**Associate profiles** icon" to "**Associate profiles** option"?

Comment (IgorCATech) [reply packages/zowe-explorer/README.md:359]: This is actually a good catch. "Icon" here is an obsolete term since the function is no longer initiated by clicking the icon that used to be part of the UI. Now, this feature is activated from the drop-down menu you get when right-clicking a profile.

**Review by crawr — APPROVED** [review]
Thanks for the quick turnaround on the zowe-docs PR!

**Comment by IgorCATech** [comment]
@lauren-li @crawr Feel free to merge this at your convenience. I know there are still 2 failing checks but it seems they are dependant on some fix...

**Review by lauren-li — APPROVED** [review]
This PR looks good to me now. Thanks @IgorCATech for the update!

**Pull request merged**
